### ISSUE TITLE
Attach-debugging in multi-user VM is basically impossible

### ISSUE THREAD

**Issue by josh-hemphill** [microsoft/vscode-dotnettools#236]
### Describe the Issue

In our environment, many users working on the same project are usually doing their work on the same VMs, which means if trying to use attach debugging there are dozens of identical process names, making it impossible to find the one that's theirs that they want to attach to.

### Expected Behavior

Some way to uniquely identify the processes like the process ID or the start directory

**Comment by tmeschter** [comment]
Moving to area-debugging as the project system isn't involved in attach scenarios.

**Comment by WardenGnaw** [comment]
@josh-hemphill You should be able to filter processes by using input box at the top of the Attach to Process prompt.

Here is searching by PID:
![image](https://github.com/microsoft/vscode-dotnettools/assets/3953714/5b70f04f-180b-4bc0-80c6-0a77fcbca1ec)

Here is searching by directory:
![image](https://github.com/microsoft/vscode-dotnettools/assets/3953714/3b99f665-11bd-4cfd-b38c-2d0c3b421d74)

Is there additional info that you would like to filter by?

**Comment by josh-hemphill** [comment]
Sorry about that, didn't notice the PIDs. I'll just have to have the program log the PID in dev runs, and I can have the devs select by that. Is there some way to surface that directly in a launch.json config? I can't see much point in setting a PID there since it will get a new one each run.

I was going to say we can't really use the directories, all the processes in our environment are running under `dotnet watch` which just gives the location of the global installation in the process list.
But I realized there is a subprocess with the correct directory displayed, it's just not in the first dozen processes or so; I scrolled through them and didn't think to search for the CWD. I wish there was some way to filter that prompt, I suppose that's something for the VSCode repo.
